src/execution_plan/ops/op_aggregate.c

```diff
@@ -163,7 +144,7 @@ static Group *_GetGroup
 
 		g = _CreateGroup(op, representative);
 
-		HashTableSetVal(op->groups, entry, g);
+		hashmap_set_with_hash(op->groups, &g, hash);
```

Comment:
In case all of our calls to `hashmap_set_with_hash` are made to store either a pointers or NULL
Then I think it would be a cleaner and simpler API to do both set and get. Also this would avoid performing memcpy by the hashmap.